src/system-health/health_checker/hardware_checker.py

```diff
@@ -102,37 +102,39 @@ def _check_fan_status(self, config):
             if not self._ignore_check(config.ignore_devices, 'fan', name, 'speed'):
                 speed = data_dict.get('speed', None)
                 speed_target = data_dict.get('speed_target', None)
-                speed_tolerance = data_dict.get('speed_tolerance', None)
+                is_under_speed = data_dict.get('is_under_speed', None)
+                is_over_speed = data_dict.get('is_over_speed', None)
                 if not speed:
                     self.set_object_not_ok('Fan', name, 'Failed to get actual speed data for {}'.format(name))
                     continue
                 elif not speed_target:
                     self.set_object_not_ok('Fan', name, 'Failed to get target speed date for {}'.format(name))
                     continue
-                elif not speed_tolerance:
-                    self.set_object_not_ok('Fan', name, 'Failed to get speed tolerance for {}'.format(name))
+                elif is_under_speed is None:
+                    self.set_object_not_ok('Fan', name, 'Failed to get under speed threshold check for {}'.format(name))
+                    continue
+                elif is_over_speed is None:
+                    self.set_object_not_ok('Fan', name, 'Failed to get over speed threshold check for {}'.format(name))
                     continue
                 else:
                     try:
                         speed = float(speed)
                         speed_target = float(speed_target)
-                        speed_tolerance = float(speed_tolerance)
-                        speed_min_th = speed_target * (1 - float(speed_tolerance) / 100)
-                        speed_max_th = speed_target * (1 + float(speed_tolerance) / 100)
-                        if speed < speed_min_th or speed > speed_max_th:
+                        if 'true' in {is_under_speed.lower(), is_over_speed.lower()}:
```

Comment:
suggest:
```
if 'true' in (is_under_speed.lower(), is_over_speed.lower()):
```

Reply:
Generally a `set` is a better structure to use with the `in` operator, but for a single use structure with a size of 2 it likely makes no difference (or perhaps would be very slightly more efficient to use a `tuple`). I'm happy to make the change, but I'm curious if there is a specific reason to suggest this way.

Reply:
set is a container only allows unique element, code like `{True, True}` looks weird. But I don't have strong opinion here, just suggestion.

Reply:
That's a fair point. Sets removing duplicates would be desirable for larger data sets, but I'll change it to a tuple here since it's more clear and single use. Good suggestion!

Reply:
Changed to tuple